packages/back-end/src/services/settings/types.ts

```diff
@@ -33,9 +35,9 @@ export interface SettingsContext {
 export type SettingsResolver<T> = (ctx: SettingsContext) => Setting<T>;
 
 export interface ScopeDefinition {
-  project?: ProjectInterface;
+  project?: ProjectSettings;
```

Comment:
I'm not sure I agree, although we can chat about this more. Since we are basically unioning all of the input interfaces, there will be a lot of conflicting fields. IMO it makes more sense to more tightly constrain the input interfaces. I don't believe there are too many cases where we need to know about the parent object, and most of those can be handled by one-offs (like denominatorMetrics, for example)